### PR TITLE
Only update the htaccess ErrorDocument links when they are not set yet

### PR DESCRIPTION
**Pull request by nickvergessen**
``` diff
diff --git a/.htaccess b/.htaccess
index 115a024..5663762 100644
--- a/.htaccess
+++ b/.htaccess
@@ -58,3 +58,22 @@ Options -Indexes
 </IfModule>
ErrorDocument 403 /../notowncloud/templates/403.php
ErrorDocument 404 /../notowncloud/templates/404.php
+
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
```

This is really not, what we intended.

@MorrisJobke @LukasReschke

### CODE REVIEW

**Comment by LukasReschke** [comment]
Makes sense :+1:

File: lib/private/setup.php [hunk 1]
```diff
@@ -402,10 +402,17 @@ public static function updateHtaccess() {
...
-		$content = "\n";
-		$content.= "ErrorDocument 403 ".\OC::$WEBROOT."/core/templates/403.php\n";//custom 403 error page
-		$content.= "ErrorDocument 404 ".\OC::$WEBROOT."/core/templates/404.php";//custom 404 error page
-		@file_put_contents($setupHelper->pathToHtaccess(), $content, FILE_APPEND); //suppress errors in case we don't have permissions for it
+		$htaccessContent = file_get_contents($setupHelper->pathToHtaccess());
+		$content = '';
+		if (strpos($htaccessContent, 'ErrorDocument 403') === false) {
+			$content.= "\nErrorDocument 403 ".\OC::$WEBROOT."/core/templates/403.php";//custom 403 error page
+		}
+		if (strpos($htaccessContent, 'ErrorDocument 403') === false) {
```
Comment (nickvergessen) [inline lib/private/setup.php]: 404 copy and paste not found

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

File: lib/private/setup.php [hunk 2]
```diff
@@ -402,10 +402,17 @@ public static function updateHtaccess() {
...
 		}
 
-		$content = "\n";
-		$content.= "ErrorDocument 403 ".\OC::$WEBROOT."/core/templates/403.php\n";//custom 403 error page
-		$content.= "ErrorDocument 404 ".\OC::$WEBROOT."/core/templates/404.php";//custom 404 error page
-		@file_put_contents($setupHelper->pathToHtaccess(), $content, FILE_APPEND); //suppress errors in case we don't have permissions for it
+		$htaccessContent = file_get_contents($setupHelper->pathToHtaccess());
+		$content = '';
+		if (strpos($htaccessContent, 'ErrorDocument 403') === false) {
+			$content.= "\nErrorDocument 403 ".\OC::$WEBROOT."/core/templates/403.php";//custom 403 error page
```
Comment (DeepDiver1975) [inline lib/private/setup.php]: can we please move the code comment one line above? THX

Comment (nickvergessen) [inline lib/private/setup.php]: pushed

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by MorrisJobke** [comment]
Jenkins is fine -> merge

**Pull request merged**